Experimental/MultiplayerUseCases/Assets/Common/Scripts/NetworkManagerUI.cs

```diff
@@ -1,76 +1,104 @@
-        [SerializeField] Color32 m_ButtonOutlineHighlightedColor;
-        [SerializeField] Color32 m_ButtonOutlineDisabledColor;
+        const string k_DefaultIP = "127.0.0.1";
+        const string k_DefaultServerListenAddress = "0.0.0.0"; //note: this is not safe for real world usage and would limit you to IPv4-only addresses, but this goes out the scope of this sample.
+        const ushort k_DefaultPort = 7979;
+
+        static class UIElementNames
+        {
+            public const string addressInputField = "IPAddressField";
+            public const string portInputField = "PortField";
+            public const string serverButton = "ServerButton";
+            public const string hostButton = "HostButton";
+            public const string clientButton = "ClientButton";
+            public const string disconnectButton = "DisconnectButton";
+            public const string quitSceneButton = "QuitSceneButton";
+        }
+
+        VisualElement m_Root;
+        TextField m_AddressInputField;
+        TextField m_PortInputField;
+        Button m_ServerButton;
+        Button m_HostButton;
+        Button m_ClientButton;
+        Button m_DisconnectButton;
+        Button m_QuitSceneButton;
+
         [SerializeField] string m_SelectionScreenSceneName;
         [SerializeField] GameObject m_ServerOnlyOverlay;
-        Button[] m_Buttons;
 
         void Awake()
         {
-            m_ServerButton.onClick.AddListener(StartServer);
-            m_HostButton.onClick.AddListener(StartHost);
-            m_ClientButton.onClick.AddListener(StartClient);
-            m_DisconnectButton.onClick.AddListener(Disconnect);
-            m_QuitSceneButton.onClick.AddListener(QuitScene);
-            m_Buttons = new Button[] { m_ServerButton, m_HostButton, m_ClientButton };
+            var uiDocument = GetComponent<UIDocument>();
+            m_Root = uiDocument.rootVisualElement;
+
+            m_AddressInputField = m_Root.Q<TextField>(UIElementNames.addressInputField);
+            m_AddressInputField.SetValueWithoutNotify(k_DefaultIP);
+
+            m_PortInputField = m_Root.Q<TextField>(UIElementNames.portInputField);
+            m_PortInputField.SetValueWithoutNotify(k_DefaultPort.ToString());
+
+            m_ServerButton = m_Root.Q<Button>(UIElementNames.serverButton);
+            m_ServerButton.RegisterCallback<ClickEvent>(StartServer);
+
+            m_HostButton = m_Root.Q<Button>(UIElementNames.hostButton);
+            m_HostButton.RegisterCallback<ClickEvent>(StartHost);
+
+            m_ClientButton = m_Root.Q<Button>(UIElementNames.clientButton);
+            m_ClientButton.RegisterCallback<ClickEvent>(StartClient);
+
+            m_DisconnectButton = m_Root.Q<Button>(UIElementNames.disconnectButton);
+            m_DisconnectButton.RegisterCallback<ClickEvent>(Disconnect);
+
+            m_QuitSceneButton = m_Root.Q<Button>(UIElementNames.quitSceneButton);
+            m_QuitSceneButton.RegisterCallback<ClickEvent>(QuitScene);
```

Comment:
@Elfi0Kuhndorf is there any reason why you're doing this "manual" initialization instead of using [the Utils](https://github.cds.internal.unity3d.com/unity/com.unity.template.multiplayer.ngo/blob/main/Assets/Scripts/Shared/UIElementsUtils.cs) that we also use in other projects (I.E: NGO template, Dedicated Server Sample)?

The utils:
1. simplify the code as you can initialize each control in one line
2. provide a common implementation across our offering
3. remove the need to add `ClickEvent evt`  in every method tied to a button, so we don't need to tamper existing code that would become harder to test through automated tests

Can you please refactor this code to use them?

Reply:
The reason why I initialized it like this is, because I saw it in another sample/template (I think it was the N4E template) done like this. But I agree, that if we can simplify the code it should be done, so I am gonna adjust it. I also did not notice this script, so thanks! I am gonna implement it.

Regarding the ClickEvent evt, are you sure it can be removed? When looking at the MainMenuView.cs of the NGO template it still uses ClickEvents for buttons in the RegisterCallback.

Reply:
> Regarding the ClickEvent evt, are you sure it can be removed?

Pretty sure about it as that's what happens in the Bootstrapper window: https://github.cds.internal.unity3d.com/unity/com.unity.template.multiplayer.ngo/blob/09e518b074550a6bd8f435f54fb0753c039ec8e1/Assets/Scripts/Editor/BootstrapperWindow.cs#L115

But you're right: some views of the NGO template aren't using the Utils to initialize buttons. It's proabably because we have a custom de-initialization flow where we unsubscribe the events from the buttons and re-set them in OnEnable.